vllm/model_executor/layers/rejection_sampler.py

```diff
@@ -164,10 +169,18 @@ def _get_accepted(
         selected_target_probs = target_probs[batch_indices, probs_indicies,
                                              draft_token_ids]
 
-        uniform_rand = torch.rand(batch_size,
-                                  k,
-                                  dtype=self.probs_dtype,
-                                  device=target_probs.device)
+
+        if any(generator is not None for generator in generators):
+            uniform_rand = torch.empty(batch_size, k, dtype=self.probs_dtype, device=target_probs.device)
+            for i, generator in enumerate(generators):
+                uniform_rand[i,:] = torch.rand(1, k, dtype=self.probs_dtype, device=target_probs.device, generator=generator)
+        else:
+
+            uniform_rand = torch.rand(batch_size,
```

Comment:
preexisting code but this could be simplified as `torch.rand_like(selected_target_probs)`

Reply:
done